src/main/java/org/spectrumauctions/sats/core/model/bvm/BMWorld.java

```diff
@@ -46,21 +48,20 @@ public BMWorld(BMWorldSetup setup, RNGSupplier rngSupplier) {
     /**
      * Gets a band with a specific name
      *
-     * @param bandName
+     * @param bandName the band name to be queried
      * @return band with this name, null if no such band in this world
      */
     public BMBand getBand(String bandName) {
-        for (BMBand band : bands) {
+        ImmutableList<BMBand> immutableBandList = ImmutableList.copyOf(bands);
```

Comment:
Why are you creating a immutable list here and don't loop over _bands_ directly? The list is never returned and out of scope after the for loop.

Reply:
I can't tell what my thought was when writing these lines, but I agree that it has no point. Thanks :)